dot/rpc/modules/state.go

```diff
@@ -245,6 +252,43 @@ func (sm *StateModule) GetPairs(_ *http.Request, req *StatePairRequest, res *Sta
 	return nil
 }
 
+// Trie RPC method returns a list of scale encoded trie.Entry{Key byte, Value byte} representing
+// all the entries in a trie for a block hash, if no block hash is given then it uses the best block hash
+func (sm *StateModule) Trie(_ *http.Request, req *StateTrieAtRequest, res *StateTrieResponse) error {
+	blockHash := sm.blockAPI.BestBlockHash()
```

Comment:
this calls looks pretty expensive and in current implementation we have to call it every time RPC method getting called. Maybe we can use if else to call it only when necessary?